ode/src/solver/BackwardEulerIvpOdeSolver.cpp

```diff
@@ -202,8 +202,9 @@ void BackwardEulerIvpOdeSolver::CalculateNextYValue(AbstractOdeSystem* pAbstract
         }
 
         counter++;
-        assert(counter < 20); // avoid infinite loops
     }
+    assert(counter < iter_limit); // avoid infinite loops
```

Comment:
Moving this assertion outside the loop makes it easier to change it into an exception later if needed.

Reply:
Yes, I think that limit should be a function parameter that defaults to (say) 20, and perhaps an exception would be better than an assert. But out of scope for this PR.